index.d.ts

```diff
@@ -107,12 +111,21 @@ declare namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action>(
-    f: Function,
+  export function run<A extends Action, F extends () => any>(
+    f: F,
     options?: {
-      args?: any[];
       failActionCreator?: ActionCreator<A>;
-      successActionCreator?: ActionCreator<A>;
+      successActionCreator?: (arg: UnwrappedReturn<F>) => A;
+      forceSync?: boolean;
+    }
+  ) : RunCmd<A>;
+
+  export function run<A extends Action, F extends Function>(
+    f: F,
+    options: {
+      args: FunctionArgs<F>;
+      failActionCreator?: ActionCreator<A>;
```

Comment:
`failActionCreator` is not typeable, for the same reason as Promise fail is not typeable.

Pseudo code:

```ts
class Promise<A> {
  constructor((resolve: (A) => void, reject: (any) => void) => void): void
}
```

This is because:

```ts
new Promise<number>((resolve, reject) => {
  throw "I can throw anything I want";
  // this is the same as
  reject("I can throw anything I want")
})
```

Reply:
Yes, that's why I did not change its type. Is there something I did not see ?